packages/discovery-provider/src/queries/get_challenges.py

```diff
@@ -65,6 +67,25 @@ def rollup_aggregates(
         step_count = sum(challenge.amount for challenge in user_challenges)
         amount = "1"
         is_complete = True
+    elif parent_challenge.id == "e":
+        sorted_challenges = sorted(
+            user_challenges, key=lambda x: x.created_at, reverse=True
+        )
+        most_recent_challenge = sorted_challenges[0]
+        if most_recent_challenge.is_complete:
+            step_count = 0
+            for challenge in sorted_challenges:
+                if challenge.current_step_count is None:
+                    continue
+                if challenge.amount < NUM_DAYS_IN_STREAK:
```

Comment:
nit:
```python
step_count += challenge.current_step_count
if challenge.amount >= NUM_DAYS_IN_STREAK:
    break
```

I also don't love the coupling with `amount` being tied to the number of days? What if we change the initial streak to 10? Or double it?

Could also use more comments probably to make this easier to grok?

Reply:
actually could just use `current_step_count` instead of amount i think.